trieste/acquisition/rule.py

```diff
@@ -1411,6 +1266,89 @@ def get_initialize_subspaces_mask(
+
+    At construction, starts in global mode using ``global_search_space`` as the search space
+    for the first step. Subsequent re-initializations use the trust region as the search space for
+    the next step.
+
+    If the previous acquisition was successful, ``global_search_space`` is used as the new
+    search space. If the previous step was unsuccessful, the search space is changed to the
+    trust region if it was global, and vice versa.
+
+    If the previous acquisition was over the trust region, the size of the trust region is
+    modified.
+
+    **Note:** The acquisition search space will never extend beyond the boundary of the
+    ``global_search_space``. For a local search, the actual search space will be the
+    intersection of the trust region and ``global_search_space``.
+    """
+
+    def __init__(
+        self,
+        global_search_space: SearchSpace,
+        beta: float = 0.7,
+        kappa: float = 1e-4,
+        min_eps: float = 1e-2,
+    ):
+        super().__init__(global_search_space, beta, kappa, min_eps)
+        self._is_global = False
+        self._initialized = False
+
+    @property
+    def eps(self) -> TensorType:
+        """The size of the search space."""
+        return self._eps
+
+    @eps.setter
+    def eps(self, eps: TensorType) -> None:
+        """Set the size of the search space."""
+        # Don't change the eps in global mode.
+        if not self._is_global:
+            self._eps = eps
+
+    def _update_bounds(self) -> None:
+        self._is_global = self._step_is_success or not self._is_global
+
+        # Use global bounds in global mode.
+        if self._is_global:
+            self._lower = self.global_search_space.lower
+            self._upper = self.global_search_space.upper
+        else:
+            super()._update_bounds()
+
+    def initialize(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        # `True` if the search space is global, else `False` if it is local.
+        # May be a scalar boolean `TensorType` instead of a `bool`.
+        #
+        # Start in global mode at construction time. Use local mode for subsequent
+        # re-initializations. Note the calls to `_update_bounds` switch the mode, so the values
+        # here are inverted.
+        self._is_global = self._initialized
+        self._initialized = True
+
+        super().initialize(models, datasets)
+
+    @inherit_check_shapes
+    def get_local_min(self, dataset: Optional[Dataset]) -> Tuple[TensorType, TensorType]:
+        if dataset is None:
+            raise ValueError("""dataset must be provided""")
+
+        # Always return the global minimum.
+        ix = tf.argmin(dataset.observations)
+        x_min = tf.gather(dataset.query_points, ix)
+        y_min = tf.gather(dataset.observations, ix)
+
+        return tf.squeeze(x_min, axis=0), tf.squeeze(y_min)
```

Comment:
@vpicheny @hstojic this is needed to match the behaviour of the previous implementation. It always used the global minimum regardless of the trust-region size or local/global mode. I have assumed we want to keep that behaviour here.